src/Transformation.tsx

```diff
@@ -98,7 +61,53 @@ function Transformation(): ReactElement {
     ],
   };
 
-  const [transformType, setTransformType] = useState<string | null>(null);
+  const transformationData: SavedTransformation[] = [
+    { name: "Build Column", content: { base: "Build Column" } },
+    { name: "Compare", content: { base: "Compare" } },
+    { name: "Count", content: { base: "Count" } },
+    { name: "Difference From", content: { base: "Difference From" } },
+    { name: "Filter", content: { base: "Filter" } },
+    { name: "Flatten", content: { base: "Flatten" } },
+    { name: "Running Sum", content: { base: "Running Sum" } },
+    { name: "Running Mean", content: { base: "Running Mean" } },
+    { name: "Running Min", content: { base: "Running Min" } },
+    { name: "Running Max", content: { base: "Running Max" } },
+    { name: "Running Difference", content: { base: "Running Difference" } },
+    { name: "Group By", content: { base: "Group By" } },
+    { name: "Pivot Longer", content: { base: "Pivot Longer" } },
+    { name: "Pivot Wider", content: { base: "Pivot Wider" } },
+    { name: "Select Attributes", content: { base: "Select Attributes" } },
+    { name: "Sort", content: { base: "Sort" } },
+    { name: "Transform Column", content: { base: "Transform Column" } },
+    { name: "Join", content: { base: "Join" } },
+    { name: "Eval", content: { base: "Eval" } },
+    { name: "Copy", content: { base: "Copy" } },
+  ];
+
+  function addTransformation(name: string, data: TransformationSaveData) {
+    // Make sure transformType is non-null
+    if (transformType === null) {
+      return;
+    }
+    // Search for the base value associated with the currently selected transformation
+    const base: BaseTransformations | undefined = transformationData.find(
+      ({ name }) => name === transformType
+    )?.content.base;
+    // Make sure the previous search was successful
+    if (base === undefined) {
+      return;
+    }
+    // Create a new transformation and add it to the list
+    // TODO: can we do this without casting?
+    const content: SavedTransformationContent = {
+      base,
+      data,
+    } as SavedTransformationContent;
+
+    const savedTransformation = { name, content };
+    const encoded = encodeURIComponent(JSON.stringify(savedTransformation));
+    createDataInteractive(name, `http://localhost:3000?transform=${encoded}`);
```

Comment:
Thought: we may want the title of the spawned data interactive (currently "CODAP Flow") to related to the name of the saved transformation somehow. If a user has created a bunch of custom transformations, minimizing them is one way to keep the workspace from getting overwhelming, but then they all look the same. It would be nice if you could easily find a minimized transformation by its title.

Reply:
This should be fixed now